src/handlers/make-render-handler.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+import type {Middleware} from "express";
+import type {Request, Response, RenderCallback} from "../types.js";
+import {extractError} from "../shared/index.js";
+import {getLogger} from "../ka-shared/index.js";
+
+/**
+ * Handle a request as a render.
+ *
+ * This method orchestrates the download and setup of a render environment
+ * and the subsequent rendering process. The downloaded code is responsible for
+ * the actual render operation.
+ *
+ * This is expected to be wrapped with express-async-handler.
+ */
+async function renderHandler(
+    renderFn: RenderCallback,
+    req: Request,
+    res: Response,
+): Promise<void> {
+    const logger = getLogger(req);
+
+    // TODO(somewhatabstract): Actually track headers and build vary header.
+    //                         Encapsulate in other code to make it easily
+    //                         tested.
+    const trackHeaderLookup = (name: string): ?string => {
+        return req.header(name);
+    };
+    // TODO(somewhatabstract): Hook in tracing (make sure that we don't leave
+    // trace sessions open on rejection (or otherwise)).
+
+    /**
+     * TODO(somewhatabstract): Currently passing the entire URL, but we
+     * want to be more specific here and define the render route better as
+     * we'll really want an absolute URL.
+     */
+    const renderURL = req.url;
+    try {
+        /**
+         * Defer this bit to the render callback.
+         */
+        const {body, status} = await renderFn(renderURL, trackHeaderLookup);
+
+        /**
+         * If the status is a redirect, we need to set the redirect header/
+         */
+        if (status.code === 301 || status.code === 302) {
+            res.header("Location", status.targetURL);
+        }
```

Comment:
I mentioned this below but I think we can just remove this - and instead add in logic for handling a map of headers in general. There are a lot of headers that we're going to need to manage during rendering and it doesn't make sense to have them be a part of the render-gateway.

Reply:
Are there a lot of headers to manage? What other headers get added to the response?

Reply:
Ones that I'm aware of from `web/request/request_handler.py`: Location, X-Frame-Options, X-Content-Type-Options - but I'm fairly certain we set other headers in other handlers - doing a quick search I see a lot of "Content-Type" and "Content-Disposition" (but maybe those will go to the REST gateway) but there are also "Cache-Control", "Pragma", "Content-Security-Policy", and "Surrogate-Control". I also see a few Set-Cookie but I think we want to move most of those to the client-side. So, in short, there may be a few but it makes more sense for us to manage those in the rendering logic rather than try to push them into the render-gateway and handle every eventuality - easier to just blanket accept headers.

Reply:
Yeah, I was planning on allowing more. The idea was to ensure that we didn't, say try to return a 301 without the header, but I think I can enforce that differently.